ext/dom/element.c

```diff
@@ -1715,6 +1715,98 @@ PHP_METHOD(Dom_Element, insertAdjacentText)
 }
 /* }}} end DOMElement::insertAdjacentText */
 
+/* https://html.spec.whatwg.org/#dom-element-insertadjacenthtml */
+PHP_METHOD(Dom_Element, insertAdjacentHTML)
+{
+	zval *where_zv;
+	zend_string *string;
+
+	dom_object *this_intern;
+	zval *id;
+	xmlNodePtr thisp;
+
+	bool created_context = false;
+
+	ZEND_PARSE_PARAMETERS_START(2, 2)
+		Z_PARAM_OBJECT_OF_CLASS(where_zv, dom_adjacent_position_class_entry)
+		Z_PARAM_STR(string)
+	ZEND_PARSE_PARAMETERS_END();
+
+	DOM_GET_THIS_OBJ(thisp, id, xmlNodePtr, this_intern);
+
+	const zend_string *where = Z_STR_P(zend_enum_fetch_case_name(Z_OBJ_P(where_zv)));
+
+	/* 1. We don't do injection sinks. */
+
+	/* 2. Let context be NULL */
+	xmlNodePtr context = NULL;
+
+	/* 3. Use the first matching item from this list: (...) */
+	switch (ZSTR_LEN(where) + ZSTR_VAL(where)[2]) {
```

Comment:
wtf

Reply:
Similar trick is done in ext/random btw if you're curious

Reply:
We really need a mechanism to solve this in a generic fashion. The RoundingMode enum uses the same trick.

Reply:
Agreed. It would be useful to be able to compare enum elements, or at least their name, by address.

Maybe we can store the address of enum elements in some struct once they are instantiated. Storing them in the `zend_class_entry` would be ideal, but I'm not sure this is doable. Case classes may help with that. Alternatively we could store them in a separate struct generated in `_arginfo.h`.